src/olympia/amo/middleware.py

```diff
@@ -95,7 +99,8 @@ class AuthenticationMiddlewareWithoutAPI(AuthenticationMiddleware):
     own authentication mechanism.
     """
     def process_request(self, request):
-        if request.path.startswith('/api/'):
+        if (request.path.startswith('/api/') and
+                not auth_view.match(request.path)):
```

Comment:
This actually wasn't required for this branch, I just forgot it in during some testing. I put the `^` and `$` in the other branch.